lib/ex_plasma/utxo.ex

```diff
@@ -66,55 +65,62 @@ defmodule ExPlasma.Utxo do
       # Create a Utxo from an Output RLP list
       iex> alias ExPlasma.Utxo
       iex> Utxo.new([<<1>>, [<<205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214, 50, 125, 202, 87, 133, 226, 40, 180>>, <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, <<13, 224, 182, 179, 167, 100, 0, 0>>]])
-      %ExPlasma.Utxo{
+      {:ok, %ExPlasma.Utxo{
         amount: 1000000000000000000,
-        blknum: 0,
+        blknum: nil,
         currency: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
-        oindex: 0,
+        oindex: nil,
         output_type: 1,
         owner: <<205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214, 50, 125,
           202, 87, 133, 226, 40, 180>>,
-        txindex: 0
-      }
+        txindex: nil
+      }}
 
       # Create a Utxo from an Input RLP Item (encoded utxo position)
       iex> alias ExPlasma.Utxo
       iex> Utxo.new(<<233, 16, 63, 218, 0>>)
-      %ExPlasma.Utxo{
-        amount: 0,
+      {:ok, %ExPlasma.Utxo{
+        amount: nil,
         blknum: 1001,
-        currency: "0x0000000000000000000000000000000000000000",
+        currency: nil,
         oindex: 0,
-        owner: "0x0000000000000000000000000000000000000000",
+        owner: nil,
         txindex: 0
-      }
+      }}
 
       # Create a Utxo from a Utxo position.
       iex> alias ExPlasma.Utxo
       iex> pos = 2000010001
       iex> Utxo.new(pos)
-      %ExPlasma.Utxo{
-        amount: 0,
+      {:ok, %ExPlasma.Utxo{
+        amount: nil,
         blknum: 2,
-        currency: "0x0000000000000000000000000000000000000000",
+        currency: nil,
         oindex: 1,
-        owner: "0x0000000000000000000000000000000000000000",
+        owner: nil,
         txindex: 1
-      }
+      }}
   """
-  @spec new(binary() | nonempty_maybe_improper_list() | non_neg_integer()) :: __MODULE__.t()
-  def new([<<output_type>>, rest_of_output]), do: new([output_type, rest_of_output])
+  @spec new(binary() | nonempty_maybe_improper_list() | non_neg_integer()) ::
```

Comment:
based on validate_input and validate_output your specs are incorrect

Reply:
ehm... marking them just atoms is incorrect way of solving this. your specs need to be specific, otherwise they're useless.

Reply:
Fixed with `__MODULE.validation_tuples()`